Extractor/Types/UAVariable.cs

```diff
@@ -212,7 +212,41 @@ public IEnumerable<UAVariable> CreateTimeseries()
                 }
                 return new[] { TimeSeries };
             }
-            return Enumerable.Empty<UAVariable>();
+            else if (NodeClass != NodeClass.Variable)
+            {
+                return Enumerable.Empty<UAVariable>();
+            }
+            else
+            {
+                return new[] { this };
+            }
+        }
+
+        public struct VariableGroups
+        {
+            public bool SourceObject;
+            public bool SourceVariable;
+            public bool DestinationObject;
+            public bool DestinationVariable;
+        }
+
+        public VariableGroups GetVariableGroups(DataTypeManager dataTypeManager)
+        {
+            var allowTsMap = dataTypeManager.AllowTSMap(this);
+            return new VariableGroups
+            {
+                // Source object if it's not a variable
+                SourceObject = NodeClass != NodeClass.Variable,
+                // Source variable if we wish to subscribe to it
+                SourceVariable = allowTsMap && NodeClass == NodeClass.Variable,
+                // Destination object if it's an object directly (through isObject)
+                // it's a mapped array, or it's not a variable.
+                DestinationObject = IsArray && Index == -1 && allowTsMap
+                    || isObject
+                    || NodeClass != NodeClass.Variable,
+                // Destination variable if allowTsMap is true and it's a variable.
+                DestinationVariable = allowTsMap && NodeClass == NodeClass.Variable,
```

Comment:
Can we make the variables more descriptive?
```suggestion
                IsSourceObject = NodeClass != NodeClass.Variable,
                // Source variable if we wish to subscribe to it
                IsSourceVariable = allowTsMap && NodeClass == NodeClass.Variable,
                // Destination object if it's an object directly (through isObject)
                // it's a mapped array, or it's not a variable.
                IsDestinationObject = IsArray && Index == -1 && allowTsMap
                    || isObject
                    || NodeClass != NodeClass.Variable,
                // Destination variable if allowTsMap is true and it's a variable.
                IsDestinationVariable = allowTsMap && NodeClass == NodeClass.Variable,
```

Reply:
I just felt that something like `VariableGroups.IsSourceObject` makes it more expressive.